src/Lykke.AlgoStore.Api/Controllers/AlgoStoreUserPermissionsController.cs

```diff
@@ -0,0 +1,158 @@
+
+        public AlgoStoreUserPermissionsController(
+            IUserRolesService userRolesService,
+            IUserPermissionsService permissionsService)
+        {
+            _userRolesService = userRolesService;
+            _permissionsService = permissionsService;
+        }
+
+        [HttpGet("getAll")]
+        [RequirePermission]
+        [SwaggerOperation("GetAllpermissions")]
+        [ProducesResponseType(typeof(List<UserPermissionModel>), (int)HttpStatusCode.OK)]
+        [ProducesResponseType(typeof(BaseErrorResponse), (int)HttpStatusCode.InternalServerError)]
+        [ProducesResponseType((int)HttpStatusCode.NotFound)]
+        public async Task<IActionResult> GetAllpermissions()
+        {
+            var result = await _permissionsService.GetAllPermissionsAsync();
+
+            return Ok(result);
+        }
+
+        [HttpGet("getById")]
+        [RequirePermission]
+        [SwaggerOperation("GetPermissionById")]
+        [ProducesResponseType(typeof(UserPermissionModel), (int)HttpStatusCode.OK)]
+        [ProducesResponseType(typeof(BaseErrorResponse), (int)HttpStatusCode.InternalServerError)]
+        [ProducesResponseType((int)HttpStatusCode.NotFound)]
+        public async Task<IActionResult> GetPermissionById(string permissionId)
+        {
+            var result = await _permissionsService.GetPermissionByIdAsync(permissionId);
+
+            return Ok(result);
+        }
+
+        [HttpGet("getByRoleId")]
+        [RequirePermission]
+        [SwaggerOperation("GetPermissionsByRoleId")]
+        [ProducesResponseType(typeof(List<UserPermissionModel>), (int)HttpStatusCode.OK)]
+        [ProducesResponseType(typeof(BaseErrorResponse), (int)HttpStatusCode.InternalServerError)]
+        [ProducesResponseType((int)HttpStatusCode.NotFound)]
+        public async Task<IActionResult> GetPermissionsByRoleId(string roleId)
+        {          
+            var result = await _permissionsService.GetPermissionsByRoleIdAsync(roleId);
+
+            return Ok(result);
+        }
+
+        [HttpPost("savePermission")]
+        [RequirePermission]
+        [SwaggerOperation("SavePermission")]
+        [ProducesResponseType(typeof(UserPermissionModel), (int)HttpStatusCode.OK)]
+        [ProducesResponseType(typeof(BaseErrorResponse), (int)HttpStatusCode.InternalServerError)]
+        [ProducesResponseType((int)HttpStatusCode.NoContent)]
+        public async Task<IActionResult> SavePermission([FromBody] UserPermissionModel permission)
+        {
+            var data = AutoMapper.Mapper.Map<UserPermissionData>(permission);
+
+            var result = await _permissionsService.SavePermissionAsync(data);
+
+            return Ok(result);
+        }
+
+        [HttpPost("assignPermission")]
+        [RequirePermission]
+        [SwaggerOperation("AssignPermissionToRole")]
+        [ProducesResponseType(typeof(UserPermissionModel), (int)HttpStatusCode.NoContent)]
```

Comment:
Is "NoContent" a type  of UserPermissionModel, or there is no model
 [ProducesResponseType(typeof(UserPermissionModel), (int)HttpStatusCode.NoContent)]
[ProducesResponseType((int)HttpStatusCode.NoContent)]
Use only one ProducesResponseType for "NoContent"